models/articles.model.js

```diff
@@ -12,7 +12,48 @@ exports.selectAllArticles = () => {
           message: "Not Found",
         });
       }
-return rows
-
+      return rows;
     });
 };
+
+exports.selectPostArticleIdComments = async (article_id, newComment) => {
+  const { username, body } = newComment;
+  if (!body.length) {
+    return Promise.reject({ status: 400, message: "Bad Request" });
+  }
+
+  const checkUsernameAndIdExists = async (article_id, username) => {
+    const dbUsername = await db.query(
+      `SELECT * FROM users WHERE username = $1`,
+      [username]
+    );
+
+    if (dbUsername.rows.length === 0) {
+      return Promise.reject({ status: 404, message: "Username not found" });
+    }
+
+    const dbArticle_Id = await db.query(
+      `SELECT * FROM articles WHERE article_id = $1`,
+      [article_id]
+    );
+
+    if (dbArticle_Id.rows.length === 0) {
+      return Promise.reject({ status: 404, message: "There is no article" });
+    }
+  };
+
+  const articleAndUsernameValid = await checkUsernameAndIdExists(
+    article_id,
+    username
+  );
+  if (articleAndUsernameValid) {
+    return checkUsernameAndIdExists(article_id, newComment);
+  }
+  const insertQuery = `INSERT INTO comments (article_id, author, body) VALUES ($1, $2, $3) RETURNING *;`;
+
+  return db
+    .query(insertQuery, [article_id, username, body])
+    .then(({ rows }) => {
+      return rows[0];
+    })
```

Comment:
This works fine, but PSQL throws it's own error when the article and username does not exist, you should catch these errors and handle these in a psql handler, it will save you about 50 lines of code :)

Reply:
This also looks suspiciously similar to another students work I've seen. Please make sure all work is your own !